core/src/com/unciv/logic/automation/ChooseBeliefsAutomation.kt

```diff
@@ -0,0 +1,185 @@
+package com.unciv.logic.automation
+
+import com.unciv.logic.city.CityInfo
+import com.unciv.logic.civilization.CivilizationInfo
+import com.unciv.logic.map.TileInfo
+import com.unciv.models.ruleset.Belief
+import com.unciv.models.ruleset.BeliefType
+import com.unciv.models.ruleset.VictoryType
+import com.unciv.models.ruleset.unique.UniqueType
+import com.unciv.models.stats.Stat
+import com.unciv.models.stats.Stats
+import kotlin.math.min
+import kotlin.random.Random
+
+object ChooseBeliefsAutomation {
+    
+    fun rateBelief(civInfo: CivilizationInfo, belief: Belief): Float {
+        var score = 0f
+        
+        for (city in civInfo.cities) {
+            for (tile in city.getCenterTile().getTilesInDistance(3)) {
+                val tileScore = beliefBonusForTile(belief, tile, city)
+                score += tileScore * when {
+                    city.workedTiles.contains(tile.position) -> 8
+                    tile.getCity() == city -> 5
+                    else -> 3
+                } * (Random.nextFloat() * 0.05f + 0.975f) 
+            }
+            
+            score += beliefBonusForCity(civInfo, belief, city) * (Random.nextFloat() * 0.1f + 0.95f)
+        }
+        
+        score += beliefBonusForPlayer(civInfo, belief) * (Random.nextFloat() * 0.3f + 0.85f)
+
+        // All of these Random.nextFloat() don't exist in the original, but I've added them to make things a bit more random.
+        
+        if (belief.type == BeliefType.Pantheon)
+            score *= 0.9f
+        
+        return score
+    }
+    
+    private fun beliefBonusForTile(belief: Belief, tile: TileInfo, city: CityInfo): Float {
+        var bonusYield = 0f
+        for (unique in belief.uniqueObjects) {
+            when (unique.placeholderText) {
```

Comment:
I wouldn't have left it at that but ensured the UniqueType can be used

Reply:
75% of this PR was made before that was introduced. I agree that it would be better, but most of these aren't on the enum yet anyway and adding all of them required a lot of additional code that is rather add in another PR